plugins/catalog-backend/src/ingestion/processors/BitbucketDiscoveryProcessor.ts

```diff
@@ -224,15 +231,23 @@ export async function readBitbucketCloud(
   return result;
 }
 
-function parseUrl(urlString: string): {
+function parseUrl(
+  urlString: string,
+  baseUrlString: string,
+): {
   projectSearchPath: RegExp;
   repoSearchPath: RegExp;
   catalogPath: string;
 } {
   const url = new URL(urlString);
-  const indexOfProjectSegment =
-    url.pathname.toLowerCase().indexOf('/projects/') + 1;
-  const path = url.pathname.substr(indexOfProjectSegment).split('/');
+  let baseUrlIndex: number;
+  if (baseUrlString) {
```

Comment:
Note, if the `baseUrl` is inferred from the `host`, then it will always be present. As such the code will always execute the `if` block and never execute the `else` block. Conversely, if we make the decision to *not* infer the `baseUrl` then both blocks would remain equally relevant.